Assets/Scripts/UserInterface/UIButton.cs

```diff
@@ -0,0 +1,70 @@
+/**
+ * Author(s): Ranjini Das
+ **/
+
+using UnityEngine;
+using UnityEngine.UI;
+using System.Collections;
+
+public class UIButton : MannBehavior
```

Comment:
You'll want to include the interface here, just to make sure you're adhering to it:

```
public class UIButton : MannBehavior, IUIButton 
{
// ...
}
```

Also note the European spelling of Behaviour (this is used in a lot of classes).